openhtf/util/testcat.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Output a JSON-encoded log output to standard out.
```

Comment:
I think this should be in `contrib/` or `bin/`, not `util/`

Reply:
[`setuptools.find_packages`](http://setuptools.readthedocs.io/en/latest/setuptools.html#using-find-packages) (used in [`setup.py`](https://github.com/google/openhtf/blob/master/setup.py#L169)) will only search for directories with `__init__.py`. Adding either `contrib/` or `bin/` to openhtf's package list will result in either `contrib` or `bin` being a top-level importable module. While it's unlikely that anyone would ever publish a [bin](https://pypi.python.org/simple/bin/) or [contrib](https://pypi.python.org/simple/contrib/) pypi module, it's not clear that we should be co-opting this part of the namespace for openhtf-specific purposes.

Reply:
I agree with @jeffherman. Unless you want to force people to go to our GitHub page and download utilities separately, stuff like this should go somewhere inside the actual package.

However I am an advocate of using runnable modules (invoke with python -m) instead of declaring entry points for setuptools to wire up executable scripts. In my anecdotal experience, the entry points mechanism is more brittle and doesn't always work with all environments. Whereas if you've gotten as far as successfully making the packager itself importable, it's more unlikely runnable modules within will fail.